solr/core/src/java/org/apache/solr/util/stats/MetricUtils.java

```diff
@@ -165,6 +169,71 @@ public static void toSolrInputDocuments(
         });
   }
 
+  /**
+   * Provides a representation of the given Dropwizard metric registry as {@link
+   * SolrPrometheusCoreExporter}-s. Only those metrics are converted which match at least one of the
+   * given MetricFilter instances.
+   *
+   * @param registry the {@link MetricRegistry} to be converted
+   * @param shouldMatchFilters a list of {@link MetricFilter} instances. A metric must match <em>any
+   *     one</em> of the filters from this list to be included in the output
+   * @param mustMatchFilter a {@link MetricFilter}. A metric <em>must</em> match this filter to be
+   *     included in the output.
+   * @param propertyFilter limit what properties of a metric are returned
+   * @param skipHistograms discard any {@link Histogram}-s and histogram parts of {@link Timer}-s.
+   * @param skipAggregateValues discard internal values of {@link AggregateMetric}-s.
+   * @param compact use compact representation for counters and gauges.
+   * @param consumer consumer that accepts produced {@link SolrPrometheusCoreExporter}-s
+   */
+  public static void toPrometheusRegistry(
```

Comment:
It's debatable that this belongs here; it doesn't seem utility-like.  It's called by one place, MetricsHandler, so that would be a decent spot.  Alternatively, given its specific-ness to Prometheus, perhaps the PrometheusResponseWriter is a better home.

Reply:
That's a fair point. Moved it to the PrometheusResponseWriter so it's known it's only used for this.